databricks/sdk/credentials_provider.py

```diff
@@ -664,6 +665,84 @@ def inner() -> Dict[str, str]:
     return inner
 
 
+# This Code is derived from Mlflow DatabricksModelServingConfigProvider
+# https://github.com/mlflow/mlflow/blob/1219e3ef1aac7d337a618a352cd859b336cf5c81/mlflow/legacy_databricks_cli/configure/provider.py#L332
+class ModelServingAuthProvider():
+    _MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+    def __init__(self):
+        self.expiry_time = -1
+        self.current_token = None
+        self.refresh_duration = 300 # 300 Seconds
+
+    def should_fetch_model_serving_environment_oauth(self) -> bool:
+        """
+        Check whether this is the model serving environment
+        Additionally check if the oauth token file path exists
+        """
+
+        is_in_model_serving_env = (os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
+                                   or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV") or "false")
+        return (is_in_model_serving_env == "true"
+                and os.path.isfile(self.__class__._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH))
+
+    def get_model_dependency_oauth_token(self, should_retry=True) -> str:
+        if self.current_token is not None and self.expiry_time > time.time():
+            return self.current_token
+        else:
+            try:
+                with open(self.__class__._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH) as f:
+                    oauth_dict = json.load(f)
+                    self.current_token = oauth_dict["OAUTH_TOKEN"][0]["oauthTokenValue"]
+                    self.expiry_time = time.time() + self.refresh_duration
+            except Exception as e:
+                # sleep and retry in case of any race conditions with OAuth refreshing
+                if should_retry:
+                    time.sleep(0.5)
+                    return self.get_model_dependency_oauth_token(should_retry=False)
+                else:
+                    raise RuntimeError(
+                        "Unable to read OAuth credentials from the file mounted in Databricks Model Serving"
+                    ) from e
+            return self.current_token
+
+    def get_databricks_host_token(self) -> Optional[Tuple[str, str]]:
+        if not self.should_fetch_model_serving_environment_oauth():
+            return None
+
+        # read from DB_MODEL_SERVING_HOST_ENV_VAR if available otherwise MODEL_SERVING_HOST_ENV_VAR
+        host = os.environ.get("DATABRICKS_MODEL_SERVING_HOST_URL") or os.environ.get(
+            "DB_MODEL_SERVING_HOST_URL")
+        token = self.get_model_dependency_oauth_token()
+
+        return (host, token)
+
+
+@credentials_strategy('model-serving', [])
+def model_serving_auth(cfg: 'Config') -> Optional[CredentialsProvider]:
+    try:
+        model_serving_auth_provider = ModelServingAuthProvider()
```

Comment:
Suggestion:
```suggestion
        model_serving_auth_provider = ModelServingAuthProvider()
        if not model_serving_auth_provider.should_fetch_model_serving_environment_oauth():
            logger.debug("model-serving: not in Databricks Model Serving, skipping")
            return None
```
Otherwise, we log a scary error message when this auth type is used outside of model serving environment.

Reply:
Good call, we dont want that error message if it is not in the mdoel serving environment.